core/src/test/java/io/opencensus/stats/MeasureMapTest.java

```diff
@@ -0,0 +1,163 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.stats;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import io.opencensus.common.Function;
+import io.opencensus.stats.Measure.DoubleMeasure;
+import io.opencensus.stats.Measure.LongMeasure;
+import io.opencensus.stats.Measurement.DoubleMeasurement;
+import io.opencensus.stats.Measurement.LongMeasurement;
+import java.util.ArrayList;
+import java.util.Iterator;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Tests for {@link MeasureMap}
+ */
+@RunWith(JUnit4.class)
+public class MeasureMapTest {
+
+  @Test
+  public void testPutDouble() {
+    ImmutableList<Measurement> expected = ImmutableList.of(
+        (Measurement) Measurement.DoubleMeasurement.create(M1, 44.4));
+    MeasureMap metrics = MeasureMap.builder().put(M1, 44.4).build();
+    assertEquals(expected.iterator(), metrics.iterator());
+  }
+
+  @Test
+  public void testPutLong() {
+    ImmutableList<Measurement> expected = ImmutableList.of(
+        (Measurement) Measurement.LongMeasurement.create(M3, 9999L),
+        Measurement.LongMeasurement.create(M4, 8888L));
+    MeasureMap metrics = MeasureMap.builder().put(M3, 9999L).put(M4, 8888L).build();
+    assertEquals(expected.iterator(), metrics.iterator());
+  }
+
+  @Test
+  public void testCombination() {
+    ImmutableList<Measurement> expected = ImmutableList.of(
+        Measurement.DoubleMeasurement.create(M1, 44.4),
+        Measurement.DoubleMeasurement.create(M2, 66.6),
+        Measurement.LongMeasurement.create(M3, 9999L),
+        Measurement.LongMeasurement.create(M4, 8888L));
+    MeasureMap metrics =
+        MeasureMap.builder().put(M1, 44.4).put(M2, 66.6).put(M3, 9999L).put(M4, 8888L).build();
+    assertEquals(expected.iterator(), metrics.iterator());
+  }
+
+  @Test
+  public void testBuilderEmpty() {
+    ImmutableList<Measurement> expected = ImmutableList.of();
+    MeasureMap metrics = MeasureMap.builder().build();
+    assertEquals(expected.iterator(), metrics.iterator());
+  }
+
+  @Test
+  public void testBuilder() {
+    ArrayList<Measurement> expected = new ArrayList<Measurement>(10);
+    MeasureMap.Builder builder = MeasureMap.builder();
+    for (int i = 1; i <= 10; i++) {
+      expected
+          .add(Measurement.DoubleMeasurement.create(makeSimpleDoubleMeasure("m" + i), i * 11.1));
+      builder.put(makeSimpleDoubleMeasure("m" + i), i * 11.1);
+      assertEquals(expected.iterator(), builder.build().iterator());
+    }
+  }
+
+  @Test
+  public void testDuplicateMeasures() {
+    // DoubleMeasure
+    MeasureMap metrics1 = MeasureMap.builder().put(M1, 1.0).build();
```

Comment:
Suggestion: You could split this test into three methods, and then the comments wouldn't be necessary.